code/datums/json_savefile.dm

```diff
@@ -79,3 +81,49 @@ GENERAL_PROTECT_DATUM(/datum/json_savefile)
 				continue
 			READ_FILE(savefile, entry_value) //we are cd'ed to the entry, so we don't need to specify a path to read from
 			region[entry] = entry_value
+
+/// Proc that handles generating a JSON file (prettified if 515 and over!) of a user's preferences and showing it to them.
+/// Requester is passed in to the ftp() and tgui_alert() procs, and account_name is just used to generate the filename.
+/// We don't _need_ to pass in account_name since this is reliant on the json_savefile datum already knowing what we correspond to, but it's here to help people keep track of their stuff.
+/datum/json_savefile/proc/export_json_to_client(mob/requester, account_name)
+	if(!istype(requester) || !path)
+		return
+
+	if(!json_export_checks(requester))
+		return
+
+	COOLDOWN_START(src, download_cooldown, (CONFIG_GET(number/seconds_cooldown_for_preferences_export) SECONDS))
```

Comment:
```suggestion
	COOLDOWN_START(src, download_cooldown, (CONFIG_GET(number/seconds_cooldown_for_preferences_export) * (1 SECONDS)))
```

We try not to abuse the macro on this too hard because it can make the code hard to read